### PR TITLE
8340731: Cleanup remaining IA64 references in hotspot code

### PR DESCRIPTION
**Pull request by offamitkumar**
This PR cleans up `IA64` entries form hotspot codebase. 

As mentioned in the JBS, I left `zero` component as-it-is: 
```
./cpu/zero/bytecodeInterpreter_zero.hpp:31:#if defined(PPC) || defined(IA64)
./os_cpu/linux_zero/os_linux_zero.cpp:350:#ifdef IA64
./os_cpu/linux_zero/os_linux_zero.cpp:351:    // IA64 has two stacks sharing the same area of memory, a normal
./os_cpu/linux_zero/os_linux_zero.cpp:360:#endif // IA64
```

Fastdebug VM Build on Mac-M1 & Linux/s390x is fine. Tests are in progress.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340731](https://bugs.openjdk.org/browse/JDK-8340731): Cleanup remaining IA64 references in hotspot code (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Andrew Haley](https://openjdk.org/census#aph) (@theRealAph - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22382/head:pull/22382` \
`$ git checkout pull/22382`

Update a local copy of the PR: \
`$ git checkout pull/22382` \
`$ git pull https://git.openjdk.org/jdk.git pull/22382/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22382`

View PR using the GUI difftool: \
`$ git pr show -t 22382`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22382.diff">https://git.openjdk.org/jdk/pull/22382.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22382#issuecomment-2499809221)
</details>

### CODE REVIEW

File: src/hotspot/share/runtime/frame.hpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/frame.hpp:167]: For now the IA32 part should remain.

File: src/hotspot/share/runtime/frame.hpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/frame.hpp:168]:
```suggestion
  // We also have relationals which allow comparing a frame to another frame's id() allowing
```

File: src/hotspot/share/runtime/javaCalls.cpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/javaCalls.cpp:121]: Not clear the comment is still valid given we no longer have sparc or ia64 support.

File: src/hotspot/share/runtime/os.hpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/os.hpp:858]:
```suggestion
  // only walk stack if %ebp is used as frame pointer.
```

File: src/hotspot/share/utilities/elfFuncDescTable.hpp
Comment (dholmes-ora) [inline src/hotspot/share/utilities/elfFuncDescTable.hpp:39]:
```suggestion
On PowerPC-64 a pointer to a function is not just a plain code address, but instead a pointer
to a so-called function descriptor (which is simply a structure containing 3 pointers).
```

**Review by dholmes-ora — COMMENTED** [review]
@offamitkumar thanks for taking up this cleanup!

Generally looks okay, a couple of nits and queries.

Thanks

File: src/hotspot/share/runtime/javaCalls.cpp
Comment (offamitkumar) [reply src/hotspot/share/runtime/javaCalls.cpp:121]: should I remove this whole comment then ?

Comment (dholmes-ora) [reply src/hotspot/share/runtime/javaCalls.cpp:121]: Yes please. I can't see anything related to "not walkable" in the thread state transition code. I checked back to Java 8.

Comment (offamitkumar) [reply src/hotspot/share/runtime/javaCalls.cpp]: done.

File: src/hotspot/share/runtime/javaCalls.cpp
Comment (offamitkumar) [reply src/hotspot/share/runtime/javaCalls.cpp]: updated :)

**Review by dholmes-ora — APPROVED** [review]
Looks good. Please wait for second review before integrating.

**Review by theRealAph — APPROVED** [review]

**Comment by offamitkumar** [comment]
all GHA test passed. Look like we are good to go. Thanks for the reviews Andrew, David. 

/integrate

**Pull request closed without merging**
